zerver/lib/stream_subscription.py

```diff
@@ -313,3 +315,94 @@ def get_subscriptions_for_send_message(
         )
     )
     return query
+
+
+def update_user_streams_subscriber_count(activate: bool, user_profile: UserProfile) -> None:
+    """
+    increment/decrement streams subscriber_count for a reactivated/deactivated user.
+    """
+    user_subscribed_streams = Stream.objects.filter(
+        recipient_id__in=get_stream_subscriptions_for_user(user_profile)
+        .filter(is_user_active=activate)
+        .values("recipient_id")
+    )
+    direction = 1 if activate else -1
+    user_subscribed_streams.update(subscriber_count=F("subscriber_count") + direction)
+
+
+def update_streams_subscriber_count(
+    direction: Literal[1, -1],
+    streams: dict[int, set[int]],
+) -> None:
+    """increment/decrement streams subscriber_count for multiple users.
+
+    direction -> 1=increment, -1=decrement
+    """
+    if len(streams) == 0:
+        return
+
+    # construct a string of tuples (stream_id, n_subscribers)
+    # to be used as the columns of the temporary table delta_table.
+    stream_delta_values = sql.SQL(", ").join(
```

Comment:
Surely there's a more idiomatic way to make an SQL array of values than this.

Reply:
done